.eslintrc.js

```diff
@@ -0,0 +1,121 @@
+    'eslint:recommended',
+    'plugin:react/recommended',
+    'plugin:import/typescript',
+    'plugin:@typescript-eslint/recommended',
+  ],
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    ecmaFeatures: {
+      jsx: true,
+    },
+    ecmaVersion: 11,
+    sourceType: 'module',
+  },
+  plugins: [
+    'import',
+    'react',
+    '@typescript-eslint',
+  ],
+  ignorePatterns: [ '*.toml', '*.json', 'node_modules/', 'dist/', '*.css', '*.scss' ],
+  rules: {
+    'linebreak-style': [ 'error', 'unix' ],
+
+    '@typescript-eslint/no-require-imports': [ 'error' ],
+
+    // 2 space indentation
+    '@typescript-eslint/indent': [ 'error', 2 ],
+
+    // Style
+    'quotes': [ 'error', 'single', { avoidEscape: true } ],
+
+    // ensures clean diffs, see https://medium.com/@nikgraf/why-you-should-enforce-dangling-commas-for-multiline-statements-d034c98e36f8
+    'comma-dangle': [ 'error', 'always-multiline' ],
+
+    // Require all imported dependencies are actually declared in package.json
+    'import/no-extraneous-dependencies': [
+      'error',
+      {
+        optionalDependencies: false,    // Disallow importing optional dependencies (those shouldn't be in use in the project)
+        peerDependencies: false,         // Disallow importing peer dependencies (that aren't also direct dependencies)
+      },
+    ],
+
+    // Require all imported libraries actually resolve (!!required for import/no-extraneous-dependencies to work!!)
+    'import/no-unresolved': [ 'error' ],
+
+    // Require an ordering on all imports
+    'import/order': ['warn', {
+      groups: ['builtin', 'external'],
+      alphabetize: { order: 'asc', caseInsensitive: true },
+    }],
+
+    // Cannot import from the same module twice
+    'no-duplicate-imports': ['error'],
+
+    // Cannot shadow names
+    'no-shadow': ['error'],
+
+    // Required spacing in property declarations (copied from TSLint, defaults are good)
+    'key-spacing': ['error'],
+
+    // Require semicolons
+    'semi': ['error', 'always'],
+
+    // Don't unnecessarily quote properties
+    'quote-props': ['error', 'consistent-as-needed'],
+
+    // No multiple empty lines
+    'no-multiple-empty-lines': ['error'],
+
+    // Max line lengths
+    'max-len': ['error', {
+      code: 120,
+      ignoreUrls: true, // Most common reason to disable it
+      ignoreStrings: true, // These are not fantastic but necessary for error messages
+      ignoreTemplateLiterals: true,
+      ignoreComments: true,
+      ignoreRegExpLiterals: true,
+    }],
+
+    // // One of the easiest mistakes to make
+    // '@typescript-eslint/no-floating-promises': ['error'],
+
+    // Don't leave log statements littering the premises!
+    'no-console': ['error'],
+
+    // Useless diff results
+    'no-trailing-spaces': ['error'],
+
+    // Must use foo.bar instead of foo['bar'] if possible
+    'dot-notation': ['error'],
+
+    // Are you sure | is not a typo for || ?
+    'no-bitwise': ['error'],
```

Comment:
Should this be `'warn'` instead of `'error'`? On the off-chance someone intentionally wants to uses a bitwise operator (although I guess it's very unlikely in web dev). Thoughts?

Reply:
lmao when would we use a bitwise operator.. if so you can always do a exclude a file in the .eslintignore or turn off a rule at the top of the file

see [this](https://stackoverflow.com/questions/34764287/turning-off-eslint-rule-for-a-specific-file)